### PR TITLE
Move Swift.php and start with travis tests - move to this app

### LINKED ISSUE #1: Move swift primary storage to an app of its own
## Current Situation
the swift implementation lives in core https://github.com/owncloud/core/blob/master/lib/private/Files/ObjectStore/Swift.php and relies on rackspace libraries which are for sure outdated.

## Todo
- [x] move swift.php into this repo
- [x] move the continuous integration in here https://github.com/owncloud/core/blob/master/Jenkinsfile#L115-L127
- [ ] require new rackspace library - mind the guzzle gap
- [ ] add makefile to allow app bundling
- [ ] create signature
- [ ] upload to marketplace
- [ ] remove from master
- [ ] remove from stable10

## Motivation
- allow own release cycle via marketplace
- shrink core
- reduce ci time

### PR DESCRIPTION
**Pull request by mrow4a**
@DeepDiver1975

fixes #1

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,14 +23,29 @@ branches:
...
+  - bash ./before_install.sh $APP_NAME $CORE_BRANCH $DB
   - cd ../core
   - php occ app:enable files_primary_swift
 
+  # Start ceph
+  - export DEBUG=on
+  - bash apps/files_primary_swift/tests/env/start-swift-ceph.sh
+  - cp apps/files_primary_swift/tests/swift.config.php config/autotest-storage-swift.config.php
+  #- cp apps/objectstore/tests/env/autotest.sh autotest.sh
+  - echo "memory_limit=2048M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini\
```
Comment (DeepDiver1975) [inline .travis.yml]: Remove the backslash at the end

File: lib/Swift.php [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Files_External\Lib\Storage;
```
Comment (DeepDiver1975) [inline lib/Swift.php]: namespace is wrong OCA\FilePrimarySwift - also add to FilePrimarySwift info.xml as namespace

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "owncloud/files_primary_swift",
+    "require": {
+    },
+    "license": "OCL"
```
Comment (DeepDiver1975) [inline composer.json:5]: please add the swift 3rdparty libs in here

File: tests/env/start-swift-ceph.sh [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+ip_address=${subnet%%/*}
+  : ${MON_IP:=${ip_address}}
+  : ${CEPH_NETWORK:=${subnet}}
+
+container=`docker run --net=host -d \
+    -e MON_IP=${MON_IP} \
+    -e CEPH_PUBLIC_NETWORK=${CEPH_NETWORK} \
+    -e CEPH_DEMO_UID=owncloud  \
+    -e RGW_CIVETWEB_PORT=${port} \
+    -e KEYSTONE_PUBLIC_PORT=${port} \
```
Comment (DeepDiver1975) [inline tests/env/start-swift-ceph.sh]: KEYSTONE_* are specific to the xenopathic containers afaik

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by DeepDiver1975** [comment]
https://travis-ci.org/owncloud/files_primary_swift/builds/264762343#L1712 - looks like a config issue ...

File: composer.json (hunk 3 above)
Comment (mrow4a) [reply composer.json:5]: what are 3rd party here?

**Pull request closed without merging**